case-payment-orders-client/src/main/java/uk/gov/hmcts/payment/casepaymentorders/client/CpoServiceClient.java

```diff
@@ -0,0 +1,110 @@
+
+    private final String url;
+    private final RestTemplate restTemplateCpoClient;
+
+    @Autowired
+    public CpoServiceClient(@Value("${case-payment-orders.api.url}") String url,
+                            @Qualifier("restTemplateCpoClient") RestTemplate restTemplateCpoClient) {
+        this.url = url;
+        this.restTemplateCpoClient = restTemplateCpoClient;
+    }
+
+    public CpoGetResponse getCasePaymentOrders(String ids, String caseIds, String pageNumber, String pageSize,
+                                               String userAuthToken, String s2sToken) {
+
+        UriComponentsBuilder builder = UriComponentsBuilder.fromHttpUrl(url + GET_CPO);
+        if (StringUtils.isNotBlank(ids)) {
+            builder.queryParam("ids", ids);
+        }
+        if (StringUtils.isNotBlank(caseIds)) {
+            builder.queryParam("case-ids", caseIds);
+        }
+        if (StringUtils.isNotBlank(pageNumber)) {
+            builder.queryParam("pageNumber", pageNumber);
+        }
+        if (StringUtils.isNotBlank(pageSize)) {
+            builder.queryParam("pageSize", pageSize);
+        }
+        HttpHeaders headers = prepareHeaders(userAuthToken, s2sToken);
+        HttpEntity<?> entity = new HttpEntity<>(headers);
+
+        try {
+            HttpEntity<String> response = restTemplateCpoClient
+                .exchange(builder.build().toUriString(),
+                          HttpMethod.GET,
+                          entity,
+                          String.class);
+
+            return cpoObjectMapper().readValue(response.getBody(), CpoGetResponse.class);
+
+        } catch (Exception e) {
+            LOG.warn("Error while retrieving Case Payment Orders: " + e.getMessage(), e);
+            if (e instanceof HttpClientErrorException
+                && HttpStatus.valueOf(((HttpClientErrorException) e).getRawStatusCode()).value() == 400) {
+                throw new CpoBadRequestException(e.getMessage(), e);
+            } else if (e instanceof HttpClientErrorException
+                && HttpStatus.valueOf(((HttpClientErrorException) e).getRawStatusCode()).is4xxClientError()) {
+                throw new CpoClientException(e.getMessage(),
+                                             HttpStatus.valueOf(((HttpClientErrorException) e)
+                                                                    .getRawStatusCode()).value(),
+                                             e);
+            } else {
+                throw new CpoInternalServerErrorException(e.getMessage(), e);
+            }
+        }
+    }
+
+    private HttpHeaders prepareHeaders(String authorization, String s2sToken) {
+        HttpHeaders headers = new HttpHeaders();
+        headers.set(CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE);
+        headers.set(SERVICE_AUTHORIZATION, s2sToken);
+        headers.set(AUTHORIZATION, authorization);
+        return headers;
+    }
+
+    protected ObjectMapper cpoObjectMapper() {
```

Comment:
Was trying to make it a Bean but then it has been picked up by other tests and broke them.